samples/java-protobuf-web-resources/src/main/java/com/example/FileServiceAction.java

```diff
@@ -0,0 +1,80 @@
+package com.example;
+
+import com.google.api.HttpBody;
+import com.google.protobuf.ByteString;
+import kalix.javasdk.Metadata;
+import kalix.javasdk.action.ActionCreationContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+// This class was initially generated based on the .proto definition by Kalix tooling.
+// This is the implementation for the Action Service described in your com.example/web_resources.proto file.
+//
+// As long as this file exists it will not be overwritten: you can maintain it yourself,
+// or delete it so it is regenerated as needed.
+
+public class FileServiceAction extends AbstractFileServiceAction {
+
+  private String defaultContentType = "text/html";
+  private Map<String, String> contentTypes = new HashMap();
+  private static final Logger LOG = LoggerFactory.getLogger(FileServiceAction.class);
+  private String baseDir = "src/main/resources/web";
+
+  public FileServiceAction(ActionCreationContext creationContext) {
+    contentTypes.put(".js","text/javascript");
+    contentTypes.put(".css","text/css");
+    contentTypes.put(".png","image/png");
+  }
+
+  private String getContentTypeByFile(String fileName) {
+    for (Map.Entry<String, String> entry : contentTypes.entrySet()) {
+      if(fileName.endsWith(entry.getKey())){
+        return entry.getValue();
+      }
+    }
+    return defaultContentType;
+  }
+
+  private  Effect<HttpBody> loadFile(String dir, String file) {
+    String fullPath = baseDir+dir+"/"+file;
+    try {
+      // tag::200-ok[]
+      byte[] byteArray = Files.readAllBytes(Paths.get(fullPath));
```

Comment:
Wait, will this actually work in a deployed app, I'd expect things to be on the class path, not unpacked into some filesystem path?

Reply:
@johanandren Yes, you are right. I was hoping to be able to add files to the docker image, like back in the days when manually writing `dockerfile`s

Reply:
Reading them from the classpath should be pretty easy though, `getClass().getClassLoader().getResourceAsStream("file")`, tricky part is if or not there should be a slash first, and if it is relative to the class or not. Note that it returns `null` if the resource is missing rather than throwing.

Reply:
Quick read up: via classloader is absolute, via `getClass().getResourceAsStream` is relative to the package of the class unless you prefix the string with slash which makes it absolute like a normal filesystem path.

Reply:
Thanks for looking it up. I tried different combinations already, but always get `null` - maybe `sbt run` messing up with things there?

Reply:
e.g. I placed the `index.html` in the same folder as the class and loading it via `getClass().getResourceAsStream("index.html").readAllBytes()` which gives an NPE.

Reply:
ok, got it. Must have missed that combinations: `getClass().getResourceAsStream("/web/index.html").readAllBytes()`